scripts/tests/java/discover_test.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+import os
+import sys
+import asyncio
+import queue
+import subprocess
+import threading
+import typing
+import argparse
+from colorama import Fore, Style
+from java.base import DumpProgramOutputToQueue
+
+
+class DiscoverTest:
+    def __init__(self, thread_list: typing.List[threading.Thread], queue: queue.Queue, cmd: [], args: str):
+        self.thread_list = thread_list
+        self.queue = queue
+        self.command = cmd
+
+        parser = argparse.ArgumentParser(description='Process discover arguments.')
+
+        parser.add_argument('command', help="Command name")
+        parser.add_argument('-n', '--nodeid', help="DNS-SD name corresponding with the given node ID", default='1')
+        parser.add_argument('-f', '--fabricid', help="DNS-SD name corresponding with the given fabric ID", default='1')
+        parser.add_argument('-p', '--paa-trust-store-path', dest='paa_trust_store_path',
+                            help="Path that contains valid and trusted PAA Root Certificates")
+
+        args = parser.parse_args(args.split())
+
+        self.command_name = args.command
+        self.nodeid = args.nodeid
+        self.fabricid = args.fabricid
+
+        logging.basicConfig(level=logging.INFO)
+
+    def TestCmdCommissionables(self):
+        java_command = self.command + ['discover', 'commissionables']
+        logging.info(f"Execute: {java_command}")
+        java_process = subprocess.Popen(
+            java_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        DumpProgramOutputToQueue(self.thread_list, Fore.GREEN + "JAVA " + Style.RESET_ALL, java_process, self.queue)
+        return java_process.wait()
+
+    def RunTest(self):
+        logging.info("Testing discovering commissionables devices")
+
+        if self.command_name == 'commissionables':
```

Comment:
consider:


```python
if self.command_name != 'commissionables':
  return 0  # TODO: why is this a success? what other values could this have?

code = self.TestCmdCommissionables()
if code != 0:
    logging.error(f"Testing command commissionables failed with error {code}")

return code
```

Changes:
- make it clear handling of different command than `commissionables` (I am wondering why we are saying that is return 0 in that case)
- decrease indent

In python we generally raise exceptions for errors. Why do we return integer codes in this method? Could we raise an exception if test fails?

Reply:
if self.command_name != 'commissionables': should not return 0, the original design is only return 0 if no return error code in any of the test. Now changed to raise exception if any of tests failed

Reply:
where is the check if self.command_name is not 'commissionables'? I.e. how do you guarantee the command?

How about adding an else branch that raises an 'unsupported command' exception?

Reply:
Acked